### PR TITLE
feat: permissions api

### PR DESCRIPTION
**Pull request by chronark**
This PR adds endpoints for all permission and roles related actions

### CODE REVIEW

**Comment by chronark** [comment]
another banger, james 😄
[CleanShot 2024-07-09 at 16.47.31@2x.png](https://files.slack.com/files-pri/T05T2KX8C3X-F07BQQNDJJG/cleanshot_2024-07-09_at_16.47.31_2x.png)

File: apps/api/src/routes/v1_keys_addPermissions.ts [hunk 1]
```diff
@@ -0,0 +1,260 @@
+import type { App } from "@/pkg/hono/app";
+import { createRoute, z } from "@hono/zod-openapi";
+
+import { rootKeyAuth } from "@/pkg/auth/root_key";
+// import { rootKeyAuth } from "@/pkg/auth/root_key";
```
Comment (perkinsjr) [inline apps/api/src/routes/v1_keys_addPermissions.ts:5]: Can we remove the commented out imports

File: apps/api/src/routes/v1_keys_addPermissions.ts [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+import { UnkeyApiError, openApiErrorResponses } from "@/pkg/errors";
+import { schema } from "@unkey/db";
+import { newId } from "@unkey/id";
+import { buildUnkeyQuery } from "@unkey/rbac";
+// import { buildUnkeyQuery } from "@unkey/rbac";
```
Comment (perkinsjr) [inline apps/api/src/routes/v1_keys_addPermissions.ts:10]: Can we remove the commented out imports

File: apps/api/src/routes/v1_keys_addRoles.ts [hunk 3]
```diff
@@ -0,0 +1,257 @@
+import type { App } from "@/pkg/hono/app";
+import { createRoute, z } from "@hono/zod-openapi";
+
+import { rootKeyAuth } from "@/pkg/auth/root_key";
+// import { rootKeyAuth } from "@/pkg/auth/root_key";
+import { UnkeyApiError, openApiErrorResponses } from "@/pkg/errors";
+import { schema } from "@unkey/db";
+import { newId } from "@unkey/id";
+import { buildUnkeyQuery } from "@unkey/rbac";
+// import { buildUnkeyQuery } from "@unkey/rbac";
```
Comment (perkinsjr) [inline apps/api/src/routes/v1_keys_addRoles.ts:10]: Can we remove the commented out imports

File: apps/api/src/routes/v1_keys_addRoles.ts [hunk 4]
```diff
@@ -0,0 +1,257 @@
+import type { App } from "@/pkg/hono/app";
+import { createRoute, z } from "@hono/zod-openapi";
+
+import { rootKeyAuth } from "@/pkg/auth/root_key";
+// import { rootKeyAuth } from "@/pkg/auth/root_key";
```
Comment (perkinsjr) [inline apps/api/src/routes/v1_keys_addRoles.ts:5]: Can we remove the commented out imports

File: packages/rbac/src/permissions.ts [hunk 5]
```diff
@@ -47,7 +47,7 @@ export const ratelimitActions = z.enum([
   "delete_namespace",
 ]);
 
-export const permissionActions = z.enum([
+export const rbacAcations = z.enum([
```
Comment (perkinsjr) [inline packages/rbac/src/permissions.ts:50]: We probably want to fix this to be`rbacActions` versus `rbacAcations`

**Review by perkinsjr — CHANGES_REQUESTED** [review]
see my comments but the one about commented out imports applys to a few more files but didn't want to continue to write it.

**Comment by chronark** [comment]
ah good, I was slowly going mad from all the copy paste and didn’t see these issues anymore

**Pull request merged**